tools/SourceKit/lib/SwiftLang/SwiftIndexing.cpp

```diff
@@ -134,10 +134,54 @@ class SKIndexDataConsumer : public IndexDataConsumer {
     if (!isRef) {
       uidAttrs = getDeclAttributeUIDs(symbol.decl);
       info.Attrs = uidAttrs;
+      if (auto *VD = dyn_cast<ValueDecl>(symbol.decl)) {
+        if (shouldOutputEffectiveAccessOfValueSymbol(symbol.symInfo)) {
+          AccessScope accessScope = VD->getFormalAccessScope();
+          UIdent AttrUID = SwiftLangSupport::getUIDForFormalAccessScope(accessScope);
+          info.EffectiveAccess = AttrUID;
+        }
+      }
     }
     return func(info);
   }
 
+  bool shouldOutputEffectiveAccessOfValueSymbol(SymbolInfo Info) {
+    SymbolKind Kind = Info.Kind;
+    SymbolSubKind SubKind = Info.SubKind;
+    switch (SubKind) {
+      case SymbolSubKind::AccessorGetter:
+      case SymbolSubKind::AccessorSetter:
+      case SymbolSubKind::SwiftAccessorWillSet:
+      case SymbolSubKind::SwiftAccessorDidSet:
+      case SymbolSubKind::SwiftAccessorAddressor:
+      case SymbolSubKind::SwiftAccessorMutableAddressor:
+      case SymbolSubKind::SwiftGenericTypeParam:
+        return false;
+      default:
+       break;
```

Comment:
```suggestion
        break;
```

Also, should this also be converted into an approach of using an allowed list, or are there too many subkinds?